server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -1171,147 +1173,205 @@ private String validateConfigurationValue(final String name, String value, final
+                    break;
+                case "netmask":
+                    errMessage = validateRangeNetmask(name, value);
+                    break;
+                case "hypervisorList":
+                    errMessage = validateRangeHypervisorList(value);
+                    break;
+                case "instanceName":
+                    errMessage = validateRangeInstanceName(value);
+                    break;
+                case "domainName":
+                    errMessage = validateRangeDomainName(value);
+                    break;
+                case "routes":
+                    errMessage = validateRangeRoutes(value);
+                    break;
+                default:
+                    errMessage = validateRangeOther(name, value, rangeOption);
+            }
+            if (errMessage == null) {
+                return null;
             }
-        } else if (type.equals(Integer.class)) {
-            final String[] options = range.split("-");
-            if (options.length != 2) {
-                final String msg = "configuration range " + range + " for " + name + " is invalid";
-                s_logger.error(msg);
-                return msg;
+            message = String.format("%1$s%2$s", message, errMessage);
+        }
+        return message;
+    }
+
+    protected String validateRangePrivateIp(String name, String value) {
+        try {
+            if (NetUtils.isSiteLocalAddress(value)) {
+                return null;
             }
-            final int min = Integer.parseInt(options[0]);
-            final int max = Integer.parseInt(options[1]);
-            final int val = Integer.parseInt(value);
-            if (val < min || val > max) {
-                s_logger.error("configuration value for " + name + " is invalid");
-                return "Please enter : " + range;
+            s_logger.error(String.format("The private IP range [%1$s] is not a site local address for the configuration variable [%2$s].", value, name));
+        } catch (final NullPointerException e) {
+            s_logger.error(String.format("Error while parsing IP address for [%s].", name));
+        }
+        return "Please enter a site local IP address.";
+    }
+
+    protected String validateRangeNetmask(String name, String value) {
+        if (NetUtils.isValidIp4Netmask(value)) {
+            return null;
+        }
+        s_logger.error(String.format("Netmask [%1$s] is not a valid netmask for configuration variable [%2$s].", value, name));
+        return "Please enter a valid netmask.";
+    }
+
+    protected String validateRangeHypervisorList(String value) {
+        final String[] hypervisors = value.split(",");
+        for (final String hypervisor : hypervisors) {
+            if (HypervisorType.getType(hypervisor) == HypervisorType.Any || HypervisorType.getType(hypervisor) == HypervisorType.None) {
+                return String.format("Please enter a valid hypervisor type.");
```

Comment:
why the `String.format()` here? did you intent to add values (like `hypervisor` or `HypervisorType`)?